internal/internal_workflow.go

```diff
@@ -1051,6 +1093,7 @@ func (d *dispatcherImpl) ExecuteUntilAllBlocked(deadlockDetectionTimeout time.Du
 	d.mutex.Unlock()
 	defer func() {
 		d.mutex.Lock()
+		d.executionState = ExecutionStateNone
```

Comment:
Is this thread safe? In other places you are concerned about the thread safety of mutating this value.